app/services/balance_transactions/creator.rb

```diff
@@ -0,0 +1,25 @@
+class BalanceTransactions::Creator
+  Params = Struct.new(:date, :comment, :amount, :user_id)
+
+  def initialize(params)
+    @params = Params.new(params[:date], params[:comment], params[:amount], params[:user_id])
+  end
+
+  def create
+    ActiveRecord::Base.transaction do
+      create_balance_transaction
+      add_transaction_to_user
+    end
+  end
+
+  private
+
+  def create_balance_transaction
+    @_transaction = BalanceTransaction.create(comment: @params[:comment])
+  end
+
+  def add_transaction_to_user
+    @_transaction.transactions.create(amount: @params[:amount], date: @params[:date].to_date,
```

Comment:
@marderer variables started with `@_` says that we don't use them in code - just for memorization
to be convenient - `@transaction`

Reply:
@AleksSenkou ok, I read that @_ are used in private area. Thanks for information.